azure-communication-ui/demo-app/src/chat/res/layout/activity_chat_launcher.xml

```diff
@@ -2,137 +2,163 @@
+        <androidx.constraintlayout.widget.ConstraintLayout
             android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:layout_marginTop="16dp"
-            android:ems="10"
-            android:hint="@string/name_hint"
-            android:inputType="textPersonName"
-            app:layout_constraintStart_toStartOf="@id/identity"
-            app:layout_constraintTop_toBottomOf="@id/identity" />
+            android:layout_height="match_parent"
+            android:paddingStart="6dp"
+            android:paddingEnd="6dp">
 
-        <EditText
-            android:id="@+id/chatThreadID"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:layout_marginTop="16dp"
-            android:ems="10"
-            android:hint="@string/chat_thread_id_hint"
-            android:inputType="textNoSuggestions"
-            app:layout_constraintStart_toStartOf="@id/userNameText"
-            app:layout_constraintTop_toBottomOf="@id/userNameText" />
-
-        <EditText
-            android:id="@+id/endPointURL"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:layout_marginTop="16dp"
-            android:ems="10"
-            android:hint="@string/end_point_url_hint"
-            android:inputType="textNoSuggestions"
-            app:layout_constraintStart_toStartOf="@id/chatThreadID"
-            app:layout_constraintTop_toBottomOf="@id/chatThreadID" />
-
-        <RadioGroup
-            android:id="@+id/javaOrKotlinContainer"
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
-            android:layout_marginTop="8dp"
-            android:orientation="horizontal"
-            app:layout_constraintStart_toStartOf="@id/endPointURL"
-            app:layout_constraintTop_toBottomOf="@id/endPointURL">
-
-            <TextView
+            <RadioButton
+                android:id="@+id/tokenFunctionRadioButton"
                 android:layout_width="wrap_content"
                 android:layout_height="wrap_content"
-                android:text="@string/launch_type_text"
-                android:textSize="17sp" />
+                android:checked="false"
+                android:text="@string/token_function_radio_button_text"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="parent" />
+
+            <EditText
+                android:id="@+id/tokenFunctionUrlText"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:ems="10"
+                android:hint="@string/token_url_hint"
+                android:inputType="textUri"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@id/tokenFunctionRadioButton" />
```

Comment:
@pavelprystinka we do not need token function for chat

Reply:
We will extend on this a bit later. Some may use token function if that is a preference.

Reply:
@pavelprystinka it is not possible to use this function anymore as the participants can not be added to chat. We do not have any tool that helps to do this.

This was already removed. I do not think this PR should add it back.

Reply:
Remoed